lib/helpers.js

```diff
@@ -226,17 +231,37 @@ function writeSourceMaps(file, destPath, callback) {
 
     var sourcemapLocation = path.relative(file.dirname, sourceMapPath);
 
-    sourcemapLocation = normalizePath(sourcemapLocation);
+    sourcemapLocation = normalizeRelpath(sourcemapLocation);
 
     comment = convert.generateMapFileComment(sourcemapLocation, commentOpts);
   }
 
   // Append source map comment
-  file.contents = appendBuffer(file.contents, comment);
+  if (comment) {
+    file.contents = appendBuffer(file.contents, Buffer.from(comment));
+  }
 
   callback(null, file, sourceMapFile);
 }
 
+function appendBuffer(buf1, buf2) {
+  var eol;
+  if (buf1.slice(-2).equals(bufCR)) {
+    eol = bufCR;
+  } else if (buf1.slice(-1).equals(bufNL)) {
+    eol = bufNL;
+  } else {
+    eol = bufEOL;
+  }
+  return Buffer.concat([buf1, buf2, eol]);
+}
+
+function normalizeRelpath(fp) {
+  // Since we only ever operate on relative paths,
+  // this utility shouldn't need to handle path roots
+  return path.posix.normalize(fp.replace(/\\/g, '/'));
+}
```

Comment:
@sttk Please review this change and verify my research/assumption is correct.

Reply:
@sttk I believe this is correct, but please inform me if incorrect. I am going to merge this PR so I can finish #33